spec/posts/index_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+RSpec.describe 'Post #Index Page', type: :feature do
+  before(:each) do
+    @user = User.create(name: 'Lilly', photo: 'https://example.jpg',
+                        bio: 'She is a teacher from Mexico', posts_counter: 10)
+    @post = Post.create(author: @user, title: 'test post', text: 'This is my first post', likes_counter: 15,
+                        comments_counter: 10)
+    @comment = Comment.create(post_id: @post.id, user_id: @user.id, text: 'test comment')
+    visit user_posts_path(@user)
+  end
+
+  it 'I can see the user\'s profile picture.' do
+    page.has_content?(@user.photo)
+  end
+
+  it 'I can see the user\'s username.' do
+    page.has_content?(@user.name)
+  end
+
+  it 'I can see the number of posts the user has written.' do
+    page.has_content?(@user.posts_counter)
+  end
+
+  it 'I can see a post\'s title.' do
+    page.has_content?(@post.title)
+  end
+
+  it 'I can see some of the post\'s body.' do
+    page.has_content?(@post.text)
+  end
+
+  it 'I can see the first comments on a post.' do
+    page.has_content?(@post.comments)
+  end
+
+  it 'I can see how many comments a post has.' do
+    page.has_content?(@post.comments_counter)
+  end
+
+  it 'I can see how many likes a post has.' do
+    page.has_content?(@post.likes_counter)
+  end
+
+  it 'I can see a section for pagination if there are more posts than fit on the view.' do
+    page.has_content?('Next')
+  end
+end

```

Comment:
- [ ] Great job adding most User post index page test cases. At the moment, you are missing one test case: `Test that redirects to the post's show page when you click on a post.` Kindly add this test case
![image](https://github.com/ritikarawat220/Blog-App/assets/62849872/44863c87-f8fe-4285-8e6f-ba9067610745)